eks-hybrid-ssm-setup.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+exec >&2
+set -eux -o pipefail
+
+HOST_ROOTFS="/.bottlerocket/rootfs"
+SSM_AGENT_PERSISTENT_STATE_DIR="${HOST_ROOTFS}/local/host-containers/control/ssm"
+SSM_AGENT_REGISTRATION="${SSM_AGENT_PERSISTENT_STATE_DIR}/registration"
+mkdir -p "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+
+extract_ssm_agent() {
+  pushd / >/dev/null 2>&1
+  dnf -yq install 'dnf-command(download)' cpio rsync
+  dnf download amazon-ssm-agent
+  rpm2cpio amazon-ssm-agent-*.rpm | cpio -idmu '*bin/amazon-ssm-agent'
+  popd >/dev/null 2>&1
+}
+
+register_hybrid_node() {
+  amazon-ssm-agent \
+    -register \
+    -region "${AWS_REGION}" \
+    -code "${SSM_ACTIVATION_CODE}" \
+    -id "${SSM_ACTIVATION_ID}" \
+    -disableSimilarityCheck
+}
```

Comment:
These need to be passed as arguments.

Reply:
If we're enabling the control container, that's what will be responsible for the SSM registration. So do we actually need this here?

Reply:
> If we're enabling the control container, that's what will be responsible for the SSM registration. So do we actually need this here?

The advantage of doing it at bootstrap time is that we can discover the node name and set the required settings before `kubelet` starts.

If we wait until the control container is running, then `kubelet` will also be running. The control container doesn't have orchestrator-specific knowledge around what to do for SSM registrations, so it wouldn't make the API calls or restart `kubelet` afterwards.

Reply:
Ah that makes sense, thanks for explaining to me!